### DISCUSSION TITLE
It's posible to detect amp mode in _document.js ?

### DISCUSSION THREAD

**Discussion by Sangdt** [vercel/next.js#20329] [Help]
So i wanted to have 2 different build for my global css file, one for amp pages and another for non-amp page because of the css rules for amp pages . [This comment](https://github.com/vercel/next.js/issues/7584#issuecomment-503376412) is really useful but it's seems like i can only import one stylesheet for the whole apps instead of a selection between 2 different builds of my css file.

I can achieve this by import css in _app.js and use the useAmp hooks but it would create a duplicate <style amp-custom> tags in document somehow....

**Comment by DedaDev** [comment]
I spend quite a lot of time figuring this out, apparently, `props._NEXT_DATA_.query` is a thing, and it is accessible in render, so the prop that you are looking for is:
`this.props.__NEXT_DATA__.query.amp`

```
class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }

  render() {
    return (
      <Html>
        <Head>
          {!this.props.__NEXT_DATA__.query.amp && <script dangerouslySetInnerHTML={{  __html: 'YOUR SCRIPT HERE OR WHATEVER' }} /> )}
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}
```
